management/src/main/scala/akka/management/AkkaManagement.scala

```diff
@@ -100,25 +119,38 @@ final class AkkaManagement(implicit system: ExtendedActorSystem) extends Extensi
   }
 
   /**
-   * Get the routes for the HTTP management endpoint.
+   * Scala API: Get the routes for the HTTP management endpoint.
    *
    * This method can be used to embed the Akka management routes in an existing Akka HTTP server.
    */
   def routes: Try[Route] = prepareCombinedRoutes(providerSettings)
 
+  // FIXME should `routes` return `Try` of throw IllegalArgumentException?
+
+  /**
+   * Java API: Get the routes for the HTTP management endpoint.
+   *
+   * This method can be used to embed the Akka management routes in an existing Akka HTTP server.
+   * @throws IllegalArgumentException if routes configured for akka management
+   */
+  def getRoutes: akka.http.javadsl.server.Route = RouteAdapter(routes.get)
+
   /**
    * Start an Akka HTTP server to serve the HTTP management endpoint.
    */
-  // FIXME make it accept config object that would have all the `withHttps`
   def start(): Future[Uri] = {
+    // FIXME API make it accept config object that would have all the `withHttps`
+    // FIXME API return CompletionStage for Java API, must use different name
```

Comment:
While we're making a breaking change for 0.50 should we add java/scala dsl packages or is this over kill?

Reply:
done in https://github.com/akka/akka-management/pull/436